File: data/imported/knowledge/slides/java/merged/snippets/snippet_017.java
@Test
    void testReroute() {
        try (var pres = new Presentation()) {
            var slide = pres.getSlides().get(0);
            var s1 = slide.getShapes().addAutoShape(ShapeType.ELLIPSE, 50, 100, 80, 80);
            var s2 = slide.getShapes().addAutoShape(ShapeType.ELLIPSE, 400, 100, 80, 80);
            var conn = slide.getShapes().addConnector(ShapeType.BENT_CONNECTOR3, 0, 0, 1, 1);
            conn.setStartShapeConnectedTo(s1);
            conn.setStartShapeConnectionSiteIndex(3);
            conn.setEndShapeConnectedTo(s2);
            conn.setEndShapeConnectionSiteIndex(1);
            conn.reroute();
            // After reroute the connector should span between the shapes
            assertThat(conn.getWidth() > 0 || conn.getHeight() > 0).isTrue();
        }
    }

File: data/imported/knowledge/slides/net/merged/snippets/snippet_017.cs
[Fact]
    public void TestReroute()
    {
        // reroute() updates connector position.
        using var pres = new Presentation();
        var slide = pres.Slides[0];
        var shapes = slide.Shapes!;
        var s1 = shapes.AddAutoShape(ShapeType.Ellipse, 50, 100, 80, 80);
        var s2 = shapes.AddAutoShape(ShapeType.Ellipse, 400, 100, 80, 80);
        var conn = shapes.AddConnector(ShapeType.BentConnector3, 0, 0, 1, 1);
        conn.StartShapeConnectedTo = s1;
        conn.StartShapeConnectionSiteIndex = 3;
        conn.EndShapeConnectedTo = s2;
        conn.EndShapeConnectionSiteIndex = 1;
        conn.Reroute();
        // After reroute the connector should span between the shapes
        (conn.Width > 0 || conn.Height > 0).Should().BeTrue();
    }